### PR TITLE
Dts 27275 lead time for change

### PR DESCRIPTION
**Pull request by hirbabar**
Logic -1
Find all Jira issues which status is contains field mapping closed status and last n release
different b/w closed date vs release date of issues
plot in x-axis based on weeks
Logic -2

Find All release version of last N months
Find All Jira issues Ids
Find all Merge request based on branchs = above tickets ids
different b/w Merge date vs release date of issues
plot in x-axis based on weeks

### CODE REVIEW

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/application/dto/FieldMappingDTO.java [hunk 1]
```diff
@@ -387,6 +381,20 @@ public class FieldMappingDTO extends BasicModel {
 
 	private List<String> jiraIssueEpicTypeKPI153;
 
+	private List<String> jiraDodKPI156;
+
+
+	private List<String> jiraIssueTypeKPI156;
+
+	@Builder.Default
```
Comment (Chittauri) [inline common/src/main/java/com/publicissapient/kpidashboard/common/model/application/dto/FieldMappingDTO.java:389]: remove builder.default

File: database-scripts/upgrade-scripts/Release-8.0-from-7.0.js [hunk 2]
```diff
@@ -3824,4 +3824,115 @@ db.field_mapping_structure.updateMany(
...
+   "fieldLabel": "Status to identify DOD",
+   "fieldType": "chips",
+   "fieldCategory": "workflow",
+   "section": "WorkFlow Status Mapping",
+   "tooltip": {
+         "definition": " Definition of Doneness. Provide any status from workflow on which DOD is considered."
+     },
+}]);
+
+db.getCollection('metadata_identifier').updateMany(
```
Comment (Chittauri) [inline database-scripts/upgrade-scripts/Release-8.0-from-7.0.js:3918]: open source has only 4 template codes, 7 to 10

File: database-scripts/rollback-scripts/Release-8.0-from-7.0-R.js [hunk 3]
```diff
@@ -511,4 +511,32 @@ db.field_mapping_structure.updateMany(
...
+]);
+// fieldMapping Structure fields for Lead time for changes in DORA tab
+db.field_mapping_structure.deleteMany({
+    "fieldName": {
+        $in: ["mergeRequestStatusKPI156", "leadTimeConfigRepoTool", "toBranchForMRKPI156", "jiraDodKPI156" , "jiraIssueTypeKPI156"]
+    }
+});
+
+db.getCollection('metadata_identifier').updateMany(
+   { "templateCode": { $in: ["4", "5", "6", "7"] } },
```
Comment (Chittauri) [inline database-scripts/rollback-scripts/Release-8.0-from-7.0-R.js:533]: opensource has template code- 7 to 10

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeForChangeServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+	/**
+	 *
+	 * @param leafNodeList
+	 * @param startDate
+	 * @param endDate
+	 * @param kpiRequest
+	 * @return
+	 */
+	@Override
+	public Map<String, Object> fetchKPIDataFromDb(List<Node> leafNodeList, String startDate, String endDate,
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeForChangeServiceImpl.java:153]: remove warnings

File: common/src/main/java/com/publicissapient/kpidashboard/common/repository/scm/MergeRequestRepositoryCustomImpl.java [hunk 5]
```diff
@@ -82,4 +90,26 @@ public List<MergeRequests> findMergeRequestList(List<ObjectId> collectorItemIdLi
 		return returnList;
 	}
 
+	@Override
```
Comment (Chittauri) [inline common/src/main/java/com/publicissapient/kpidashboard/common/repository/scm/MergeRequestRepositoryCustomImpl.java:93]: add comment

File: database-scripts/rollback-scripts/Release-8.0-from-7.0-R.js [hunk 6]
```diff
@@ -511,4 +511,32 @@ db.field_mapping_structure.updateMany(
         $set: { "fieldCategory": "workflow" }
     }
 
-)
\ No newline at end of file
+)
+
+
+db.kpi_master.bulkWrite([
+// delete Lead time for change KPI (156)
```
Comment (Chittauri) [inline database-scripts/rollback-scripts/Release-8.0-from-7.0-R.js:518]: remove bulk write

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/application/dto/FieldMappingDTO.java (hunk 1 above)
Comment (hirbabar) [reply common/src/main/java/com/publicissapient/kpidashboard/common/model/application/dto/FieldMappingDTO.java:389]: needed

**Review by hargupta15 — DISMISSED** [review]

**Review by eswbogol — APPROVED** [review]

**Review by brahmanand1 — DISMISSED** [review]

**Review by eswbogol — APPROVED** [review]

**Review by brahmanand1 — APPROVED** [review]

**Pull request merged**